float64[] color_weight
int32[] corner_number
